db/dcrpg/queries.go

```diff
@@ -1354,6 +1354,76 @@ func RetrieveAddressTxnOutputWithTransaction(db *sql.DB, address string, current
 	return outputs, nil
 }
 
+// retrieveTicketsByDate fetches the tickets in the current ticketpool by the
+// purchase date.
+func retrieveTicketsByDate(db *sql.DB, maturityBlock, groupBy int64) (*dbtypes.PoolTicketsData, error) {
+	rows, err := db.Query(internal.SelectTicketsByPurchaseDate, groupBy, maturityBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	tickets := new(dbtypes.PoolTicketsData)
+	for rows.Next() {
+		var immature, live, timestamp uint64
+		var price, total float64
+		err = rows.Scan(&timestamp, &price, &immature, &live)
+		if err != nil {
+			return nil, fmt.Errorf("retrieveTicketsByDate %v", err)
+		}
+
+		tickets.Time = append(tickets.Time, timestamp)
+		tickets.Immature = append(tickets.Immature, immature)
+		tickets.Live = append(tickets.Live, live)
+
+		// Returns the average value of a ticket depending on the grouping mode used
+		price = price * 100000000
```

Comment:
Why convert from coin to atoms, then back to coin?  But in general, when converting either way, use dcrutil.  In this case it would have been dcrutil.NewAmount.

Reply:
We need to get the average price and still maintain the 8 decimal places for our average price.
Converting it to atoms, get the average price then format to have a max of 8 decimal places did seem like the easiest way.

Reply:
Hmm, it seems like the max of 8 decimal places is a display/string processing issue.  The average quantity is just a statistic, not an actual single quantity accounted directly in the block chain.  So there's no need to keep the decimals to 8 except for display consistency

Reply:
Yes, display consistency is why I had to do all that.

Reply:
Display consistency was the reason.

Reply:
In that case, the float64AsDecimalParts and/or amountAsDecimalParts template functions take care of the formatting.  There's really no reason to do the `price = price * 100000000`, then `dcrutil.Amount(price/total).ToCoin()` when `price` starts in units of coin. We just need `price/total` and the template function handles formatting.

Reply:
But in the query we are getting the `SUM(tickets.price) AS price,` that match a given grouping provided. From the sum thats where we need to get the average of the summed prices.

Reply:
Doing this on the frontend might slow down the browser loading the ticketpool page everytime new data is available. I think this is cheaper to do it the backend while we fetch the summed prices.

Reply:
In the worst case scenario we may end up having 40k records I think using float64AsDecimalParts and/or amountAsDecimalParts template to do the formatting may make small devices browser unresponsive when loading data in the ticketpool page.